src/frontends/pytorch/src/utils_quantize.hpp

```diff
@@ -0,0 +1,156 @@
+
+    QuantizedPtNode(const QuantizedPtNodeType type,
+                    const NodeContext& context,
+                    const std::shared_ptr<ov::Node> input,
+                    const std::shared_ptr<ov::Node> scale,
+                    const std::shared_ptr<ov::Node> zero_point,
+                    const std::shared_ptr<ov::Node> axis = nullptr)
+        : PtFrameworkNode(context.get_decoder(), {input}, 1, false),
+          type(type),
+          scale(scale),
+          zero_point(zero_point),
+          axis(axis) {
+        ov::op::util::FrameworkNodeAttrs attrs = get_attrs();
+        if (type == QuantizedPtNodeType::QUANTIZE_PER_TENSOR) {
+            attrs[quantized_node_type_key] = quantize_per_tensor;
+        } else if (type == QuantizedPtNodeType::QUANTIZE_PER_CHANNEL) {
+            FRONT_END_OP_CONVERSION_CHECK(axis != nullptr, "quantize_per_channel requires axis to be provided.");
+            attrs[quantized_node_type_key] = quantize_per_channel;
+        } else {
+            FRONT_END_OP_CONVERSION_CHECK(false, "Unknown QuantizedPtNodeType: ", type);
+        }
+        set_attrs(attrs);
+    }
+
+    QuantizedPtNode(const QuantizedPtNodeType type,
+                    const NodeContext& context,
+                    const ov::Output<ov::Node> input,
+                    const ov::Output<ov::Node> scale,
+                    const ov::Output<ov::Node> zero_point)
+        : QuantizedPtNode(type, context, input.get_node_shared_ptr(), scale.get_node_shared_ptr(), zero_point.get_node_shared_ptr(), nullptr) {}
+
+
+    QuantizedPtNode(const QuantizedPtNodeType type,
+                    const NodeContext& context,
+                    const ov::Output<ov::Node> input,
+                    const ov::Output<ov::Node> scale,
+                    const ov::Output<ov::Node> zero_point,
+                    const ov::Output<ov::Node> axis_)
+        : QuantizedPtNode(type, context, input.get_node_shared_ptr(), scale.get_node_shared_ptr(), zero_point.get_node_shared_ptr(), axis_.get_node_shared_ptr()) {}
+
+    const std::shared_ptr<ov::Node> get_scale() {
+        return scale;
+    }
+    const std::shared_ptr<ov::Node> get_zero_point() {
+        return zero_point;
+    }
+    const std::shared_ptr<ov::Node> get_axis() {
+        return axis;
+    }
+    const QuantizedPtNodeType get_type() {
+        return type;
+    }
+    void set_scale(std::shared_ptr<ov::Node> scale_) {
+        scale = scale_;
+    }
+    void set_zero_point(std::shared_ptr<ov::Node> zero_point_) {
+        zero_point = zero_point_;
+    }
+
+private:
+    const QuantizedPtNodeType type;
+    std::shared_ptr<ov::Node> scale;
+    std::shared_ptr<ov::Node> zero_point;
+    std::shared_ptr<ov::Node> axis;
+};
+
+/**
+ * Quantizes input node with the given parameters. Returns a shared pointer to the new QuantizedPtNode.
+*/
+std::shared_ptr<ov::Node> quantize(const NodeContext& context,
```

Comment:
I don't think you need methods for `std::shared_ptr<Node>` and for `Output<Node>`. Methods that accept outputs should be sufficient.

Reply:
Also since you pass `NodeContext` to this function you shouldn't need the pointer to `Node` itself from outside, so I would prefer to return `Output<Node>` from these functions.

Reply:
Done!